js/style/style.js

```diff
@@ -647,7 +676,27 @@ Style.prototype = util.inherit(Evented, {
             this._handleErrors(validateStyle.filter, 'querySourceFeatures.filter', params.filter, true);
         }
         var source = this.getSource(sourceID);
-        return source && source.querySourceFeatures ? source.querySourceFeatures(params) : [];
+        return source ? Source._querySourceFeatures(source, params) : [];
+    },
+
+    addSourceType: function (name, sourceFactoryFn, workerSourceURL, callback) {
```

Comment:
@lucaswoj this could also be `(name, { create: (id, options, dispatcher, onChange, callback) => void, ?workerSourceURL }, callback)`.  Any preference between the two signatures?